pkg/puller/puller.go

```diff
@@ -294,16 +300,28 @@ func (p *Puller) histSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 			return
 		}
 		if s > cur {
-			p.logger.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur, "peer_address", peer)
+			p.logger.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur, "duration", time.Since(start), "peer_address", peer)
 			return
 		}
+
+		ctx, cancel := context.WithTimeout(ctx, recalcPeersDur)
```

Comment:
Should this bee syncPeersTimeout?

I would rather call it histSyncTimeout as this is only for historical syncing.